saleor/tests/e2e/shop/utils/preparing_shop.py

```diff
@@ -1,44 +1,286 @@
-from ...channel.utils import create_channel
-from ...shipping_zone.utils.shipping_method import create_shipping_method
-from ...shipping_zone.utils.shipping_method_channel_listing import (
+import uuid
+
+from ...channel.utils import create_channel, update_channel
+from ...shipping_zone.utils import (
+    create_shipping_method,
     create_shipping_method_channel_listing,
+    create_shipping_zone,
+    update_shipping_price,
+)
+from ...taxes.utils import (
+    create_tax_class,
+    get_tax_configurations,
+    update_country_tax_rates,
+    update_tax_configuration,
 )
-from ...shipping_zone.utils.shipping_zone import create_shipping_zone
-from ...warehouse.utils import create_warehouse
-
-
-def prepare_shop(e2e_staff_api_client):
-    warehouse_data = create_warehouse(e2e_staff_api_client)
-    warehouse_id = warehouse_data["id"]
-    channel_slug = "test"
-    warehouse_ids = [warehouse_id]
-    channel_data = create_channel(
-        e2e_staff_api_client,
-        slug=channel_slug,
-        warehouse_ids=warehouse_ids,
+from ...warehouse.utils import create_warehouse, update_warehouse
+from ..utils.shop_update_settings import update_shop_settings
+
+
+def create_and_update_tax_rates(e2e_staff_api_client, tax_rates):
+    created_tax_classes = {}
+    for tax_rate_info in tax_rates:
+        tax_class_data = create_tax_class(
+            e2e_staff_api_client,
+            tax_rate_info["name"],
+            [
+                {
+                    "countryCode": tax_rate_info["country_code"],
+                    "rate": tax_rate_info["rate"],
+                }
+            ],
+        )
+        created_tax_classes[tax_rate_info["type"]] = {
+            "id": tax_class_data["id"],
+            "rate": tax_rate_info["rate"],
+        }
+
+        update_country_tax_rates(
+            e2e_staff_api_client,
+            tax_rate_info["country_code"],
+            [{"taxClassId": tax_class_data["id"], "rate": tax_rate_info["rate"]}],
```

Comment:
For updating the country rate we should skip sending`{"taxClassId": tax_class_data["id"]`

> updateTaxClassRates: [TaxClassRateInput!]!
> List of tax rates per tax class to update. When {taxClass: id, rate: null} is passed, it deletes the rate object for given taxClass ID. When {rate: Int} is passed without a tax class, it updates the default tax class for this country.